readthedocs/config/config.py

```diff
@@ -511,58 +507,24 @@ def formats(self):
     @property
     def python(self):
         """Python related configuration."""
-        return self._config.get('python', {})
-
-    @property
-    def python_version(self):
-        """Python version."""
-        return self._config['python']['version']
-
-    @property
-    def pip_install(self):
-        """True if the project should be installed using pip."""
-        return self._config['python']['pip_install']
-
-    @property
-    def install_project(self):
-        """True if the project should be installed."""
-        if self.pip_install:
-            return True
-        return self._config['python']['setup_py_install']
-
-    @property
-    def extra_requirements(self):
-        """Extra requirements to be installed with pip."""
-        if self.pip_install:
-            return self._config['python']['extra_requirements']
-        return []
-
-    @property
-    def use_system_site_packages(self):
-        """True if the project should have access to the system packages."""
-        return self._config['python']['use_system_site_packages']
-
-    @property
-    def use_conda(self):
-        """True if the project use Conda."""
-        return self._config.get('conda') is not None
+        requirements = self._config['requirements_file']
+        self._config['python']['requirements'] = requirements
```

Comment:
Are we sure that `self._config['python']` will always exists _and_ be a dictionary? In the init we are not defining it.

Reply:
Also, I think this method may return `None`, right?

Reply:
Yeah, we always have a valid dict for `self._config['python']`. How is that? We always need to call to `validate` before using any of this properties, there is a docstring about that. The method always returns a valid object